### PR TITLE
doc/architecture: "Edit HA Auth" (one of several)

### PR DESCRIPTION
**Pull request by zdover23**
Edit "High Availability Authentication" in doc/architecture.rst.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/architecture.rst [hunk 1]
```diff
@@ -235,30 +235,34 @@ authenticated connections.
...
-legitimate messages, as long as the user's secret key is not divulged before it
-expires.
-
-To use ``cephx``, an administrator must set up users first. In the following
-diagram, the ``client.admin`` user invokes  ``ceph auth get-or-create-key`` from
-the command line to generate a username and secret key. Ceph's ``auth``
-subsystem generates the username and key, stores a copy with the monitor(s) and
-transmits the user's secret back to the ``client.admin`` user. This means that 
-the client and the monitor share a secret key.
+A user or actor invokes a Ceph client to contact a monitor. Unlike Kerberos,
```
Comment (anthonyeleven) [inline doc/architecture.rst:238]: actor?

File: doc/architecture.rst [hunk 2]
```diff
@@ -235,30 +235,34 @@ authenticated connections.
...
+each monitor can authenticate users and distribute keys, which means that there
+is no single point of failure and no bottleneck when using ``cephx``. The
+monitor returns an authentication data structure that is similar to a Kerberos
+ticket. This authentication data structure contains a session key for use in
+obtaining Ceph services. The session key is itself encrypted with the user's
+permanent secret key, which means that only the user can request services from
+the Ceph Monitor(s). The client then uses the session key to request services
```
Comment (anthonyeleven) [inline doc/architecture.rst:245]: Monitors

File: doc/architecture.rst [hunk 3]
```diff
@@ -235,30 +235,34 @@ authenticated connections.
...
+from the monitor, and the monitor provides the client with a ticket that
```
Comment (anthonyeleven) [inline doc/architecture.rst:246]: singular here, plural above

File: doc/architecture.rst [hunk 4]
```diff
@@ -235,30 +235,34 @@ authenticated connections.
...
+server in the cluster. 
+
+Like Kerberos tickets, ``cephx`` tickets expire. An attacker cannot use an
+expired ticket or session key that has been obtained surreptitiously. This form
+of authentication prevents attackers who have access to the communications
+medium from creating bogus messages under another user's identity and prevents
+attackers from altering another user's legitimate messages, as long as the
+user's secret key is not divulged before it expires.
+
+An administrator of a Ceph cluster must set up users before using ``cephx``.
```
Comment (anthonyeleven) [inline doc/architecture.rst:259]: /of a Ceph cluster//

**Review by anthonyeleven — APPROVED** [review]

File: doc/architecture.rst (hunk 1 above)
Comment (zdover23) [reply doc/architecture.rst:238]: Again, you have ratified my reservations. "[A]ctor" is history.

File: doc/architecture.rst (hunk 2 above)
Comment (zdover23) [reply doc/architecture.rst:245]: Accepted.

File: doc/architecture.rst (hunk 3 above)
Comment (zdover23) [reply doc/architecture.rst:246]: I've made all of these nouns plural.

File: doc/architecture.rst (hunk 4 above)
Comment (zdover23) [reply doc/architecture.rst:259]: Accepted.

File: doc/architecture.rst (hunk 1 above)
Comment (anthonyeleven) [reply doc/architecture.rst]: Rad!

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/53585 - Reef backport
https://github.com/ceph/ceph/pull/53586 - Quincy backport

**Pull request merged**